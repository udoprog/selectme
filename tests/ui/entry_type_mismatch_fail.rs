fn main() {}

#[selectme::main]
async fn missing_semicolon_or_return_type() {
    Ok(())
}

#[selectme::main]
async fn missing_return_type() {
    return Ok(());
}

#[selectme::main]
async fn extra_semicolon() -> Result<(), ()> {
    Ok(());
}
