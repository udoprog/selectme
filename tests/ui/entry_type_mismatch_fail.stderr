error[E0308]: mismatched types
 --> tests/ui/entry_type_mismatch_fail.rs:5:5
  |
5 |     Ok(())
  |     ^^^^^^- help: consider using a semicolon here: `;`
  |     |
  |     expected `()`, found enum `Result`
  |
  = note: expected unit type `()`
                  found enum `Result<(), _>`

error[E0308]: mismatched types
  --> tests/ui/entry_type_mismatch_fail.rs:10:12
   |
10 |     return Ok(());
   |            ^^^^^^ expected `()`, found enum `Result`
   |
   = note: expected unit type `()`
                   found enum `Result<(), _>`

error[E0308]: mismatched types
  --> tests/ui/entry_type_mismatch_fail.rs:14:46
   |
14 |   async fn extra_semicolon() -> Result<(), ()> {
   |  ______________________________________________^
15 | |     Ok(());
   | |           - help: consider removing this semicolon
16 | | }
   | |_^ expected enum `Result`, found `()`
   |
   = note:   expected enum `Result<(), ()>`
           found unit type `()`
